src/rust/src/lib.rs

```diff
@@ -2,7 +2,72 @@
 // 2.0, and the BSD License. See the LICENSE file in the root of this repository
 // for complete details.
 
+/// Returns the value of the input with the most-significant-bit copied to all
+/// of the bits.
+fn duplicate_msb_to_all(a: u8) -> u8 {
+    (1 - (a >> 7)).wrapping_sub(1)
+}
+
+/// This returns 0xFF if a < b else 0x0000, but does so in a constant time
+/// fashion.
```

Comment:
The problem is size-independent... Moving to `u8` just changes the constants, but the problem stays the same.

Reply:
Indeed, I'm adapting the logic from OpenSSL and adding a test that simply tries everything in the `(a, b)` state space, which is practical now that they're `u8`